components/brave_wallet/browser/eth_tx_controller.cc

```diff
@@ -669,6 +669,51 @@ void EthTxController::SetGasPriceAndLimitForUnapprovedTransaction(
   std::move(callback).Run(true);
 }
 
+void EthTxController::SetGasFeeAndLimitForUnapprovedTransaction(
+    const std::string& tx_meta_id,
+    const std::string& max_priority_fee_per_gas,
+    const std::string& max_fee_per_gas,
+    const std::string& gas_limit,
+    SetGasFeeAndLimitForUnapprovedTransactionCallback callback) {
+  if (max_priority_fee_per_gas.empty() || max_fee_per_gas.empty() ||
+      gas_limit.empty()) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  std::unique_ptr<EthTxStateManager::TxMeta> tx_meta =
+      tx_state_manager_->GetTx(tx_meta_id);
+  if (!tx_meta || tx_meta->status != mojom::TransactionStatus::Unapproved) {
+    std::move(callback).Run(false);
+    return;
+  }
+
+  auto* tx1559 = reinterpret_cast<Eip1559Transaction*>(tx_meta->tx.get());
```

Comment:
need to check `EthTransaction::Type() == 2` before continuing and we should also add a test case that trying to call this function for non eip1559 transaction and we should expect failure

Reply:
Good call, fixed in https://github.com/brave/brave-core/pull/10332/commits/df3df47c0a637144c7ff37269037e5b3bdfd01f0.